File: src/main/java/com/braggbnb109/service/AvailabilityService.java
package com.braggbnb109.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb109.domain.Availability;
import com.braggbnb109.dto.AvailabilityDTO;
import com.braggbnb109.dto.AvailabilitySearchDTO;
import com.braggbnb109.dto.AvailabilityPageDTO;
import com.braggbnb109.dto.AvailabilityConvertCriteriaDTO;
import com.braggbnb109.service.GenericService;
import com.braggbnb109.dto.common.RequestDTO;
import com.braggbnb109.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AvailabilityService extends GenericService<Availability, Integer> {

	List<Availability> findAll();

	ResultDTO addAvailability(AvailabilityDTO availabilityDTO, RequestDTO requestDTO);

	ResultDTO updateAvailability(AvailabilityDTO availabilityDTO, RequestDTO requestDTO);

    Page<Availability> getAllAvailabilitys(Pageable pageable);

    Page<Availability> getAllAvailabilitys(Specification<Availability> spec, Pageable pageable);

	ResponseEntity<AvailabilityPageDTO> getAvailabilitys(AvailabilitySearchDTO availabilitySearchDTO);
	
	List<AvailabilityDTO> convertAvailabilitysToAvailabilityDTOs(List<Availability> availabilitys, AvailabilityConvertCriteriaDTO convertCriteria);

	AvailabilityDTO getAvailabilityDTOById(Integer availabilityId);







}

File: braggbnb109_ui/src/services/AvailabilityService.js
import http from "../http-common"; 

class AvailabilityService {
  getAllAvailabilitys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/availability/availabilitys`, searchDTO);
  }

  get(availabilityId) {
    return this.getRequest(`/availability/${availabilityId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/availability?field=${matchData}`, null);
  }

  addAvailability(data) {
    return http.post("/availability/addAvailability", data);
  }

  update(data) {
  	return http.post("/availability/updateAvailability", data);
  }
  
  uploadImage(data,availabilityId) {
  	return http.postForm("/availability/uploadImage/"+availabilityId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AvailabilityService();
